### PR TITLE
Updated gateway contract

### PR DESCRIPTION
**Pull request by susruth**
* Corrected the contract's name
* Changed the constructor to accept minimum darkpool size

### CODE REVIEW

File: contracts/Gateway.sol [hunk 1]
```diff
@@ -11,10 +11,16 @@ contract RepublicProtocol is Ownable {
...
+  function Gateway
+  (   address _token, 
+      address _darkNodeRegistry, 
+      address _traderRegistry,
+      uint256 _darkpoolSize
+  ) public {
     token = _token;
     darkNodeRegistry = _darkNodeRegistry;
     traderRegistry = _traderRegistry;
+    darkpoolSize = _darkpoolSize;
```
Comment (loongy) [inline contracts/Gateway.sol]: Being pedantic, it should be `_darkPoolSize` ;)

**Comment by loongy** [comment]
If we include the `_darkPoolSize` parameter, then we should also have a function `updateDarkPoolSize`.

**Comment by susruth** [comment]
It was included in the previous pull request #6

File: contracts/Gateway.sol (hunk 1 above)
Comment (susruth) [reply contracts/Gateway.sol]: done

**Pull request merged**